pkg/projectfile/projectfile.go

```diff
@@ -437,17 +439,40 @@ var persistentProject *Project
 
 // Parse the given filepath, which should be the full path to an activestate.yaml file
 func Parse(configFilepath string) (*Project, *failures.Failure) {
-	dat, err := ioutil.ReadFile(configFilepath)
+	projectDir := filepath.Dir(configFilepath)
+	files, err := ioutil.ReadDir(projectDir)
 	if err != nil {
-		return nil, failures.FailIO.Wrap(err)
+		return nil, failures.FailIO.Wrap(err, locale.Tl("err_project_readdir", "Could not check for project files in your project directory."))
 	}
 
-	project := Project{}
-	err = yaml.Unmarshal([]byte(dat), &project)
-	project.path = configFilepath
+	project, fail := parse(configFilepath)
+	if fail != nil {
+		return nil, fail
+	}
 
-	if err != nil {
-		return nil, FailParseProject.New(locale.T("err_project_parse", map[string]interface{}{"Error": err.Error()}))
+	re, _ := regexp.Compile(`activestate.(\w+).yaml`)
+	for _, file := range files {
+		match := re.FindStringSubmatch(file.Name())
+		if len(match) == 0 {
+			continue
+		}
+
+		// If an OS keyword was used ensure it matches our runtime
+		l := strings.ToLower
+		keyword := l(match[1])
+		if (keyword == l(sysinfo.Linux.String()) || keyword == l(sysinfo.Mac.String()) || keyword == l(sysinfo.Windows.String())) &&
+			keyword != l(sysinfo.OS().String()) {
+			logging.Debug("Not merging %s as we're not on %s", file.Name(), sysinfo.OS().String())
+			continue
+		}
+
+		secondaryProject, fail := parse(filepath.Join(projectDir, file.Name()))
+		if fail != nil {
+			return nil, fail
+		}
+		if err := mergo.Merge(project, *secondaryProject, mergo.WithAppendSlice); err != nil {
+			return nil, failures.FailMarshal.Wrap(err, locale.Tl("err_merge_project", "Could not merge {{.V0}} into your activestate.yaml", file.Name()))
+		}
 	}
```

Comment:
nit: If you gather the files using a glob, you can be sure that all the returned file names are intended to be merged.
https://github.com/ActiveState/cli/compare/dgreen/exp-asy_split#diff-6a2389bce029e4a55df04d9f53423c45R20-R46

This should be faster (Readdir is heavy-handed, imo) and the code should be simpler. Using split is safe because the glob guarantees that the filenames will have three sections, but I'm not strongly suggesting to not use regex.